components/raftstore/src/store/worker/region.rs

```diff
@@ -671,13 +672,20 @@ where
                 end_key,
             } => {
                 fail_point!("on_region_worker_destroy", true, |_| {});
-                // try to delay the range deletion because
-                // there might be a coprocessor request related to this range
-                self.ctx
-                    .insert_pending_delete_range(region_id, &start_key, &end_key);
-
-                // try to delete stale ranges if there are any
-                self.ctx.clean_stale_ranges();
+                // If `region_id` is 0, it's used for deleting extra range from splitting.
+                if region_id == 0 {
+                    self.ctx.cleanup_range(
```

Comment:
Why delete it immediately? There may still be running queries.

Reply:
It's a mistake. I read `region work` code and find it can be handled like applying a different range snapshot.